src/codewhisperer/util/telemetryHelper.ts

```diff
@@ -486,18 +551,81 @@ export class TelemetryHelper {
         telemetry.codewhisperer_clientComponentLatency.emit({
             codewhispererRequestId: this.firstResponseRequestId,
             codewhispererSessionId: this.sessionId,
-            codewhispererFirstCompletionLatency: this.sdkApiCallEndTime - this.sdkApiCallStartTime,
-            codewhispererEndToEndLatency: this.firstSuggestionShowTime - this.invokeSuggestionStartTime,
-            codewhispererAllCompletionsLatency: this.allPaginationEndTime - this.sdkApiCallStartTime,
+            codewhispererFirstCompletionLatency:
+                this.sdkApiCallEndTime - CodeWhispererSessionStates.instance.sdkApiCallStartTime,
+            codewhispererEndToEndLatency:
+                this.firstSuggestionShowTime - CodeWhispererSessionStates.instance.invokeSuggestionStartTime,
+            codewhispererAllCompletionsLatency:
+                this.allPaginationEndTime - CodeWhispererSessionStates.instance.sdkApiCallStartTime,
             codewhispererPostprocessingLatency: this.firstSuggestionShowTime - this.sdkApiCallEndTime,
-            codewhispererCredentialFetchingLatency: this.sdkApiCallStartTime - this.fetchCredentialStartTime,
-            codewhispererPreprocessingLatency: this.fetchCredentialStartTime - this.invokeSuggestionStartTime,
+            codewhispererCredentialFetchingLatency:
+                CodeWhispererSessionStates.instance.sdkApiCallStartTime -
+                CodeWhispererSessionStates.instance.fetchCredentialStartTime,
+            codewhispererPreprocessingLatency:
+                CodeWhispererSessionStates.instance.fetchCredentialStartTime -
+                CodeWhispererSessionStates.instance.invokeSuggestionStartTime,
             codewhispererCompletionType: 'Line',
             codewhispererTriggerType: this.triggerType,
             codewhispererLanguage: runtimeLanguageContext.getLanguageContext(languageId).language,
-            credentialStartUrl: this.startUrl,
+            credentialStartUrl: AuthUtil.instance.startUrl,
             codewhispererUserGroup: CodeWhispererUserGroupSettings.getUserGroup().toString(),
         })
-        this.resetClientComponentLatencyTime()
+    }
+
+    public sendUserModificationEvent(suggestion: AcceptedSuggestionEntry, percentage: number) {
+        client
+            .sendTelemetryEvent({
+                telemetryEvent: {
+                    userModificationEvent: {
+                        sessionId: suggestion.sessionId ? suggestion.sessionId : 'undefined',
```

Comment:
Is it necessary to send it when session id is undefined?

Reply:
no and in fact we shouldn't send if sessionId is undefined.

Reply:
```suggestion.sessionId ? suggestion.sessionId : 'undefined'``
I don't think this is even needed. sessionId will in the worst be an emptpy string but not undefined.

Reply:
Another thing is sessionId and requestId don't make sense for the userModificationEvent, and this function is also not get called currently, I will simply remove this method.